lib/spack/spack/spec.py

```diff
@@ -1958,16 +1957,29 @@ def concretize(self, tests=False):
                     for pcond, patch_list in dependency.patches.items():
                         if dspec.spec.satisfies(pcond):
                             for patch in patch_list:
-                                patches.append(patch.sha256)
+                                patches.append(patch)
             if patches:
-                mvar = dspec.spec.variants.setdefault(
-                    'patches', MultiValuedVariant('patches', ())
-                )
-                mvar.value = mvar.value + tuple(patches)
-                # FIXME: Monkey patches mvar to store patches order
-                p = getattr(mvar, '_patches_in_order_of_appearance', [])
-                mvar._patches_in_order_of_appearance = list(
-                    dedupe(p + patches))
+                all_patches = spec_to_patches.setdefault(id(dspec.spec), [])
```

Comment:
For future maintainability, can you add a comment here to clarify what's going on. Something like "extending dependency package patches with patches coming from its dependents" would make it easier for someone to pick up this code later.

Reply:
The loop initializer has a comment like this:

> # Apply patches required on dependencies by depends_on(..., patch=...)

are you saying that it should be clarified like:

> # Add patches for packages that are required by their dependents (that appear in depends_on)

(or perhaps replacing that comment with yours)

?

Reply:
I just missed that comment when I was reading/didn't go up far enough. I think that's fine.